cmd/format-config-v1.go

```diff
@@ -839,15 +839,22 @@ func checkFormatXLValues(formatConfigs []*formatConfigV1) (int, error) {
 	return -1, nil
 }
 
-// checkFormatXL - verifies if format.json format is intact.
+// checkFormatXL - verifies if format.json format is intact. If an
+// error is found, the disk that caused the first error is also
+// returned in the error message.
 func checkFormatXL(formatConfigs []*formatConfigV1) error {
-	if _, err := checkFormatXLValues(formatConfigs); err != nil {
-		return err
-	}
-	if err := checkJBODConsistency(formatConfigs); err != nil {
-		return err
+	var err error
+	errCheckCall := func(f func([]*formatConfigV1) (int, error)) {
+		if err == nil {
+			if i, e := f(formatConfigs); e != nil {
+				err = fmt.Errorf("error on disk %d: %v", i, e)
```

Comment:
While returning here we can write the `index+1` to point at the right disk number instead of the slice index number.